dts/arm/nordic/nrf52840.dtsi

```diff
@@ -13,8 +13,18 @@
 		};
 	};
 
-	flash0: flash@0 {
-		reg = <0x00000000 DT_FLASH_SIZE>;
+	flash-controller@0x4001E000 {
+			compatible = "nrf,nrf52-flash-controller";
+			reg = <0x4001E000 0x550>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash {
+				compatible = "flash";
+				reg = <0x00000000 DT_FLASH_SIZE>;
+				write-block-size = <4>;
```

Comment:
e.g. mcuboot, and mcuboot interface (#747) needs to know this value. For me It doesn't mater whether this is flash-controller's property  value or flash memory's property (anyway these entities always existing together). But in my opinion f`lash` is right node for this property.